src/app/query/bitcoin/ordinals/utils.ts

```diff
@@ -85,3 +63,65 @@ export function getTaprootAddress(index: number, keychain: HDKey, network: Netwo
 
   return payment.address;
 }
+
+/**
+ * Ordinals contain arbitrary data. When retrieving an ordinal, we should
+ * classify it into one of the types below, indicating that it is handled
+ * appropriately. Ordinals of types not ready to be handled by the app should be
+ * classified as other.
+ */
+const supportedOrdinalTypes = ['image', 'other'] as const;
+
+type SupportedOrdinalType = (typeof supportedOrdinalTypes)[number];
+
+interface BaseOrdinalInfo {
+  /**
+   * The kind of ordinal as classified by this app. Different kinds of ordinals
+   * require different treatment (e.g., images vs documents).
+   */
+  type: SupportedOrdinalType;
+
+  /**
+   * Title which can be rendered in the UI alongside the ordinal.
+   */
+  title: string;
+
+  /**
+   * A link to a detailed techincal description about the ordinal.
+   */
+  infoUrl: string;
+}
+
+interface ImageOrdinalInfo extends BaseOrdinalInfo {
+  /**
+   * A URL of where the image can be downloaded from.
+   */
+  type: 'image';
+  content: string;
+}
+
+interface OtherOrdinalInfo extends BaseOrdinalInfo {
+  /**
+   * A URL of where the image can be downloaded from.
+   */
```

Comment:
Personally I'd use comments a little more sparingly than this. The prop name/context would probably make this clear its an img url, and here it describes the type property that _isn't_ a url. Below is a type string?